### ISSUE TITLE
TOTP/ HOTP 2FA for login page

### ISSUE THREAD

**Issue by ammnt** [AdguardTeam/AdGuardHome#1667]
Hello,

please add TOTP/ HOTP 2FA for login page.

Thank you.

Cheers!

**Comment by owenvoke** [comment]
Related to this, maybe even [WebAuthn](https://webauthn.guide) support. 🤔 I guess TOTP can be easier to implement sometimes, but WebAuthn would be a great addition.

**Comment by michaelb-ae** [comment]
i only ask, any news on this ? (totp) and why not add NtpSec (https://github.com/AdguardTeam/AdGuardHome/issues/2267) support (client at least) to check the hour (for being sure the otp is correct).

**Comment by ameshkov** [comment]
Nope, no news. But since there're relatively many upvotes, it's time to at least assign it to a milestone.

**Comment by thijsjek** [comment]
I have DOH/dot  internet facing, so I can use it on my android private DNS. Henceforth it is showing the login page at these subdomains (using reverse proxy) 
2fa would be a good security add-on.

**Comment by BirdInFire** [comment]
> Nope, no news. But since there're relatively many upvotes, it's time to at least assign it to a milestone.

Any news 3 year later it become quite a long wait...

**Comment by owenvoke** [comment]
@BirdInFire, looks like this is planned for milestone [`v0.109`](https://github.com/AdguardTeam/AdGuardHome/milestone/29), so can't see this happening anytime soon. 😬 It's been nearly 2 years that `v0.108` has been in beta.

---

On a related note, [Passkeys](https://fidoalliance.org/passkeys) might be a good option to add as well if this ever gets implemented. 🤷🏻

**Comment by BirdInFire** [comment]
> @BirdInFire, looks like this is planned for milestone [`v0.109`](https://github.com/AdguardTeam/AdGuardHome/milestone/29), so can't see this happening anytime soon. 😬 It's been nearly 2 years that `v0.108` has been in beta.
> 
> On a related note, [Passkeys](https://fidoalliance.org/passkeys) might be a good option to add as well if this ever gets implemented. 🤷🏻

I lost hope and use authentik before it (for password less login), but their app completely ignore authorization header so I must log twice, and the trick of very long ttl only work if you remember password

**Comment by shukryshuk** [comment]
any news ? my login page is exposed to public 🫠🫠

**Comment by ainar-g** [comment]
No news, and you don't need 2FA to make your web interface unavailable from the outside.  You can ask around in Discussions to find ways that fit your case.

**Comment by pictosun** [comment]
Any update concerning 2FA for AGH? Especially when using it via a VPS server this would be a nice feature to have.

**Comment by BirdInFire** [comment]
> No update after 4 years...

The world move to passkey and adguard don't have otp on home .... Up-to-date security I tell.

**Comment by sorcer1122** [comment]
Also waiting for this

**Comment by pictosun** [comment]
Any update concerning 2FA login features for AGH?
